internal/common/baseclient/baseclient.go

```diff
@@ -26,6 +26,16 @@ type Client interface {
 	//IIFO: OPTIONAL
 	MakePrediction() (shared.PredictionInfo, error)
 	ReceivePredictions(receivedPredictions shared.PredictionInfoDict) error
+
+	//IITO: COMPULSORY
+	RequestGift() int
+	OfferGifts(giftRequestDict shared.GiftDict) (shared.GiftDict, error)
+	AcceptGifts(receivedGiftDict shared.GiftDict) (shared.GiftInfoDict, error)
+	UpdateGiftInfo(acceptedGifts map[shared.ClientID]shared.GiftInfoDict) error
+
+	//TODO: THESE ARE NOT DONE yet, how do people think we should implement the actual transfer?
+	SendGift(receivingClient shared.ClientID, amount int) error
+	ReceiveGift(sendingClient shared.ClientID, amount int) error
```

Comment:
`uint` I guess

on the note of the transfer impl. -- the server acts as a middle man? The server would call a "GetGifts" function that describes a list of gifts to other clients.

Reply:
The server middle-man impl. does mean more work, but it prevents the two problems above. I feel policing at a level of programming implementation is fair (as opposed to policing cheating resources which is part of the game).

Reply:
A rejection of the gift can be later updated to the client

Reply:
Should this gate this PR or should I make another PR implementing the transfer of resources?

Reply:
new PR is better I think! Unless you think it won't clutter this one too much :)

Reply:
Moved to issue